### PR TITLE
Core v7 update

### PR DESCRIPTION
**Pull request by yvesgoeleven**
### Things to do

-  Upgrade the packaging to use the new csproj packaging. 
   * [x] Add a `Directory.Build.targets` file in `/src` that [contains the following](https://github.com/Particular/NServiceBus/blob/develop/src/Directory.Build.targets).
   * [x] Add `    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>` to the project(s) that should generate nuget packages
   * [x] Add a description eg. `<Description>A testing framework for NServiceBus</Description>` , see https://github.com/Particular/NServiceBus.Testing/blob/develop/src/NServiceBus.Testing/NServiceBus.Testing.csproj#L14
   * [x] Delete `./packaging`
   * [x] Remove reference to NuGetPackager
   * [x] [Make sure to use gitversiontask 4.0.0-beta001 or higher](https://github.com/Particular/NServiceBus.NLog/blob/develop/src/NServiceBus.NLog/NServiceBus.NLog.csproj#L35)
   * ~~Multitarget netstandard 2.0 if trivial. If not raise an issue in this repo to track a future update etc~~
   * [x] make sure to mark build-only dependencies with `PrivateAssets="All"`
-  [x] [Upgrade to latest core v7 alpha](https://github.com/Particular/NServiceBus.NLog/blob/develop/src/NServiceBus.NLog/NServiceBus.NLog.csproj#L33) 
    * [x] Decide if using a wildcard ref is what you want or not. Some maintainers (rabbit) have chosen not to since a bad core build could start to fail their build
    * [x] Note that there has been some changes to acceptance test cleanup. Ping @Particular/nservicebus-maintainers if you run into touble (we'll evolve a FAQ on this issue as we go)
- ~~do not upgrade the NUnit.TestAdapter package to the latest stable (3.8) but use 3.8-alpha1 instead~~
- [x] [Bump major in gitversion.yml](https://github.com/Particular/NServiceBus/blob/develop/GitVersion.yml#L2)
- ~~No need to touch the buildserver unless a linux build is demed a must for beta~~
- ~~If inspections fail, update to latest in repostandards~~
- ~~ApprovalTests/PublicApiGenerator doesn't yet run on linux so make sure to only run them on net452~~
- [x] Push a Alpha build to myget 
- [x] Make sure that the build is using the latest build template (StandardBuildV3 as of writing this)
- [ ] Make doco updates as need
   * [ ] Add new snippets for vNext.Major
       -  [ ] If possible multi target the sample against netstandard 2.0
   * [ ] Add new samples for  vNext.Major and smoke test them
   * [ ] Upgrade guide if there is breaking changes
- [x] Remove usages of "ApiApprover" and use the "ApprovalTests" package instead. See https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core.Tests/NServiceBus.Core.Tests.csproj#L23 and https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core.Tests/API/APIApprovals.cs
- [x] use this fix to make approval tests run without exception when running in R#: https://github.com/Particular/NServiceBus.Testing/pull/111

### CODE REVIEW

**Comment by SeanFeldman** [comment]
API has changed -- requires review/approval.

File: src/AcceptanceTests/NServiceBus.AzureServiceBus.AcceptanceTests.csproj [hunk 1]
```diff
@@ -11,12 +11,12 @@
...
+    <PackageReference Include="WindowsAzure.ServiceBus" Version="4.1.3" />
     <PackageReference Include="Microsoft.WindowsAzure.ConfigurationManager" Version="3.2.3" />
-    <PackageReference Include="NServiceBus.AcceptanceTesting" Version="6.3.1" />
-    <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="6.3.1" />
-    <PackageReference Include="NServiceBus" Version="6.3.1" />
-    <PackageReference Include="NUnit" Version="3.6.1" />
+    <PackageReference Include="NServiceBus.AcceptanceTesting" Version="7.0.0-alpharelease0051" />
+    <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="7.0.0-alpharelease0051" />
+    <PackageReference Include="NServiceBus" Version="7.0.0-alpharelease0051" />
+    <PackageReference Include="NUnit" Version="3.7.1" />
```
Comment (SeanFeldman) [inline src/AcceptanceTests/NServiceBus.AzureServiceBus.AcceptanceTests.csproj]: Has the issue of having assertions in handlers in ATTs was addressed? NUnit 3.7.1 will work only if assertions are **not** made in handlers (different threads). PR #574 has already attempted that with NSB Core v6 and it didn't work, had to wait for NUnit 3.8.1.

File: src/AcceptanceTests/Receiving/When_incoming_message_lock_token_is_lost_in_send_atomic_with_receive_mode.cs [hunk 2]
```diff
@@ -68,9 +65,7 @@ public Receiver()
             {
                 EndpointSetup<DefaultServer>(config =>
                 {
-#pragma warning disable 618
-                    config.Recoverability().DisableLegacyRetriesSatellite();
-#pragma warning restore 618
```
Comment (SeanFeldman) [inline src/AcceptanceTests/Receiving/When_incoming_message_lock_token_is_lost_in_send_atomic_with_receive_mode.cs]: Finally 🙂

File: src/Transport/NServiceBus.AzureServiceBus.csproj [hunk 3]
```diff
@@ -20,9 +20,10 @@
   <ItemGroup>
     <None Include="..\..\packaging\nuget\nservicebus.azureservicebus.nuspec" Link="nservicebus.azureservicebus.nuspec" />
     <PackageReference Include="Janitor.Fody" Version="1.4.1" />
-    <PackageReference Include="WindowsAzure.ServiceBus" Version="4.0.0" />
+    <PackageReference Include="NServiceBus.Newtonsoft.Json" Version="1.2.0-alpha0017" />
+    <PackageReference Include="WindowsAzure.ServiceBus" Version="4.1.3" />
```
Comment (SeanFeldman) [inline src/Transport/NServiceBus.AzureServiceBus.csproj]: Do we update ASB client version or keep it at 4.0.0?

**Comment by SeanFeldman** [comment]
@yvesgoeleven [this list](https://github.com/Particular/DotNetCoreLaunch/issues/7) needs to be implemented as well to have NSB Core v7 implementation.

File: src/AcceptanceTests/NServiceBus.AzureServiceBus.AcceptanceTests.csproj (hunk 1 above)
Comment (yvesgoeleven) [inline src/AcceptanceTests/NServiceBus.AzureServiceBus.AcceptanceTests.csproj]: Should be released this weekend https://github.com/nunit/nunit/issues/2376

Comment (SeanFeldman) [reply src/AcceptanceTests/NServiceBus.AzureServiceBus.AcceptanceTests.csproj]: I've closed #574 as we no longer need it. This PR essentially replaces it.

**Comment by SeanFeldman** [comment]
Green build. Merged.

**Pull request merged**